core/src/main/java/bio/terra/pearl/core/service/participant/ParticipantUserService.java

```diff
@@ -44,4 +59,21 @@ public Optional<ParticipantUser> findOne(String username, EnvironmentName enviro
         return dao.findOne(username, environmentName);
     }
 
+    public Optional<ParticipantUser> findOneByShortcode(String shortcode) {
+        return dao.findByProperty("shortcode", shortcode);
+    }
+
+    // creates an "ACC"-prefixed shortcode for each participant user that's missing one
+    public List<UUID> createMissingShortcodes() {
+        List<ParticipantUser> users = dao.findAllWithMissingShortcode();
+
+        users.forEach(participantUser -> {
+            String shortcode = shortcodeService.generateShortcode("ACC", dao::findOneByShortcode);
+            participantUser.setShortcode(shortcode);
+            dao.update(participantUser);
+        });
```

Comment:
this will be a bunch of queries...but also this is one-off throwaway code. happy to switch to a batch query though

Reply:
yeah, it should be in the low thousands, which as you say, isn't great, but fine for a one-time thing (and probably a few thousand queries fewer than happen during data export 😛 ).  I would just add a comment that we expect to delete this method in a follow-up PR where we add the non-nullable constraint to shortcode

Reply:
Good idea, just filed https://broadworkbench.atlassian.net/browse/JN-1318 and referenced it in the code comment